src/ring_theory/polynomial/eisenstein.lean

```diff
@@ -242,6 +242,58 @@ lemma cyclotomic_comp_X_add_one_is_eisenstein_at [hp : fact p.prime] :
     exact nat.prime.not_dvd_one hp.out (dvd.intro k (hk.symm)),
   end }
 
+lemma cyclotomic_prime_pow_comp_X_add_one_is_eisenstein_at [hp : fact p.prime] (n : ℕ) :
+  ((cyclotomic (p ^ (n + 1)) ℤ).comp (X + 1)).is_eisenstein_at 𝓟 :=
+{ leading :=
+  begin
```

Comment:
Does it make sense to have a lemma that says proves the goal assuming the polynomial is monic? You are now basically inlining the proof of that lemma. But I think it's a common case that deserves a separate lemma.

Reply:
Do you mean a lemma that says that if `f.monic` then `f.leading_coeff ∉ 𝓟`?

Reply:
I've added `polynomial.monic.leading_coeff_not_mem`.

Reply:
I guess that's also a useful lemma. But I was thinking of a lemma that says: a monic polynomial is eisenstein at P if blabla.
I think blabla is easier then all the work you are doing now, so it would be a good API lemma.

Reply:
I've added `polynomial.monic.is_eisenstein_at_of_mem_of_not_mem`. I am not sure this is what you meant, but I don't see a simpler characterization.